collated/main/A0139915W.md

```diff
@@ -0,0 +1,1430 @@
+# A0139915W
+###### \java\seedu\savvytasker\commons\events\ui\TaskPanelSelectionChangedEvent.java
+``` java
+
+/**
+ * Represents a selection change in the Task List Panel
+ */
+public class TaskPanelSelectionChangedEvent extends BaseEvent {
+
+
+    private final ReadOnlyTask newSelection;
+
+    public TaskPanelSelectionChangedEvent(ReadOnlyTask newSelection){
+        this.newSelection = newSelection;
+    }
+
+    @Override
+    public String toString() {
+        return this.getClass().getSimpleName();
+    }
+
+    public ReadOnlyTask getNewSelection() {
+        return newSelection;
+    }
+}
+```
+###### \java\seedu\savvytasker\commons\util\SmartDefaultDates.java
+``` java
+/**
+ * Helper functions for handling dates.
+ * @author A0139915W
+ */
+
+public class SmartDefaultDates {
+    
+    private Date startDateTime;
+    private Date endDateTime;
+    private final Calendar calendar;
+    private final Calendar today;
+    
+    /**
+     * 
+     * @param startDateTime Starting date time
+     * @param endDateTime Ending date time
+     * @throws InvalidDateException If endDateTime is earlier than startDateTime
```

Comment:
I don't see a `throws Exception` statement. Also, avoid using `@param` if the arguments are self-explanatory.